test/parallel/test-https-immutable-options.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const https = require('https');
+const tls = require('tls');
+
+const dftProtocol = {};
+tls.convertNPNProtocols([ 'http/1.1', 'http/1.0' ], dftProtocol);
+
+const opts = { foo: 'bar' };
+const server1 = https.createServer(opts);
+
+assert.deepStrictEqual(opts, { foo: 'bar' });
+assert.strictEqual(server1.NPNProtocols.compare(dftProtocol.NPNProtocols), 0);
```

Comment:
@mscdex I think we can't, because `NPNProtocols` hasn't been cached yet. 2 calls to `tls.convertNPNProtocols` won't return the same reference.